watertap/core/util/initialization.py

```diff
@@ -136,3 +138,44 @@ def assert_no_degrees_of_freedom(blk):
 
     """
     check_dof(blk, True)
+
+
+def interval_initializer(blk, feasibility_tol=1e-6):
+    """
+    Improve the initialization of ``blk`` utilizing interval arithmetic.
+
+    Keyword Arguments:
+        blk : block to initialize
+        feasibility_tol : tolerance to use for FBBT (default: 1e-6)
+
+    Returns:
+        None
+
+    """
+
+    bound_cache = ComponentMap()
+
+    for v in blk.component_data_objects(Var, active=True, descend_into=True):
+        bound_cache[v] = v.bounds
+
+    fbbt(blk, feasibility_tol=feasibility_tol, deactivate_satisfied_constraints=False)
+
+    for v, bounds in bound_cache.items():
+        if v.value is None:
+            # set to the bound nearer 0
```

Comment:
This comment appears to be incorrect - it is setting the value, not the bounds.

Also, I don't think `0` is a good value to use here. For one, I suspect this indicates this variable is unbounded so we have no idea about its value, and `0`  will probably cause someone to get evaluation errors. So, I am not really sure what the "best" value would be.

I would suggest logging an `INFO` or even an `ERROR` message to let the user know they should manually address this (or add some bounds), and use either an `EPS` value or maybe even 1.

Reply:
I am sympathetic to the idea we should raise an error here, though nothing else in our software stack will check this. And note that Ipopt will 0-initialize such values anyways.

The comment refers to the following fact: If FBBT is able to prove that `lb <= ub <=0`, then the below code will change the initial value to the `ub`. Similarly, if we prove `0 <= lb <= ub`, then the below code will change the initial value to the `lb`.

Maybe we should pick a different value, however, and not raise the error.

Reply:
To clarify - I wasn't suggesting actually raising an exception, but just to log a message at the `INFO` or `ERROR` level to ensure the user sees it (i.e. making sure they are aware that this might be an issue and that they should check).